lib/src/react16_suggestors/dependency_override_updater.dart

```diff
@@ -14,69 +14,105 @@
+
+    // This needs to be kept track of because the patches are applied all at
+    // once, so even though we will add the dependency_override key the file
+    // won't be aware until all the patching is done.
+    bool shouldAddDependencyOverrideKey = dependencyOverrideSectionKey == null;
+
+    // Each override has its own object to keep track of what the dependency
+    // override is and what it needs to be overridden with.
+    // TODO update these versions to the dev branch after major release.
+    final reactDependencyOverride = DependencyOverrideItem(
+        'react',
+        '  react:\n'
+            '    git:\n'
+            '      url: https://github.com/cleandart/react-dart.git\n'
+            '      ref: 5.0.0-wip\n');
+
+    final overReactDependencyOverride = DependencyOverrideItem(
+        'over_react',
+        '  over_react:\n'
+            '    git:\n'
+            '      url: https://github.com/Workiva/over_react.git\n'
+            '      ref: 3.0.0-wip\n');
+
+    final dependenciesToUpdate = [
+      reactDependencyOverride,
+      overReactDependencyOverride
+    ];
+
+    final parsedYamlMap = loadYaml(contents);
+
+    for (var i = 0; i < dependenciesToUpdate.length; i++) {
+      final dependency = dependenciesToUpdate[i].dependency;
+      final overrideString = dependenciesToUpdate[i].overrideString;
+
+      if (fileContainsDependencyOverride(
+          dependency: dependency, yamlContent: parsedYamlMap)) {
+        // The regex that can be used to find the location of the override.
+        final dependencyRegex = getDependencyRegEx(
+            dependency: dependency, yamlContent: parsedYamlMap);
+        final dependencyMatch = dependencyRegex.firstMatch(contents);
+
+        if (dependencyMatch != null) {
+          // startPoint is needed because a new line would be added to the
+          // top of the patch. This controls whether we need to move the
+          // starting line up one or keep it the default.
+          int startPoint = dependencyMatch.start;
+
+          if (sourceFile.getLine(dependencyOverrideSectionKey.end) !=
+              sourceFile.getLine(dependencyMatch.start - 1)) {
+            startPoint = dependencyMatch.start - 1;
+          }
+
+          yield Patch(sourceFile,
+              sourceFile.span(startPoint, dependencyMatch.end), overrideString);
+        }
       } else {
-        yield Patch(
-            sourceFile,
-            sourceFile.span(span, span),
-            '\n'
-            'dependency_overrides:\n'
-            '$dependencyOverrides');
+        int insertionOffset;
+        int insertionLine;
+
+        // If the section already exists, insert new dependencies directly
+        // below the key. If not, insert at the end of the file.
+        if (dependencyOverrideSectionKey != null) {
+          insertionLine =
+              sourceFile.getLine(dependencyOverrideSectionKey.end) + 1;
+        } else {
+          insertionLine = sourceFile.lines - 1;
```

Comment:
Does this mess things up if the file doesn't have a trailing newline, but content on its last line?

Reply:
Not that I can tell. That was a good point so I added a test to check and manually ran it on that case, and it handled it as expected!